sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java

```diff
@@ -60,16 +56,17 @@
 
     @Before
     public void prepare() throws IOException {
+        tableStats = new TableStats();
         sqlExecutor = SQLExecutor.builder(clusterService)
             .enableDefaultTables()
+            .setTableStats(tableStats)
             .addView(new RelationName("doc", "v2"), "select a, x from doc.t1")
             .addView(new RelationName("doc", "v3"), "select a, x from doc.t1")
             .build();
-        tableStats = new TableStats();
     }
 
     private LogicalPlan plan(String statement) {
-        return plan(statement, sqlExecutor, clusterService, tableStats);
+        return withMemoryLimit(ByteSizeUnit.MB.toBytes(8), () -> sqlExecutor.logicalPlan(statement));
```

Comment:
Why is this limited to 8MB?

Reply:
Currently the plan method allocates around 5-7 MB. The intention here is to have a regression check to make sure we don't increase the allocations a lot and also to make us aware of what various methods are allocating. For example, the deleted block below caused an extra ~40MB to be allocated: (probably due to `getFunctions()`)

```
PlannerContext context = sqlExecutor.getPlannerContext(clusterService.state());
        AnalyzedRelation relation = sqlExecutor.analyze(statement);
        LogicalPlanner logicalPlanner = new LogicalPlanner(
            getFunctions(),
            tableStats,
            () -> clusterService.state().nodes().getMinNodeVersion()
        );
        SubqueryPlanner subqueryPlanner = new SubqueryPlanner((s) -> logicalPlanner.planSubSelect(s, context));
```

Reply:
I see thanks, yes wording was a bit confusing. I'd suggest `assertMaxBytesAllocated`.

Reply:
@seut Changed it.